### PR TITLE
Cleanup eslint config

### PR DESCRIPTION
**Pull request by fregante**
Setup for https://github.com/pixiebrix/pixiebrix-extension/issues/536
Includes parts of https://github.com/pixiebrix/pixiebrix-extension/pull/513

This PR ensures that the `lint` script actually runs without errors.

Before moving onto adding further plugins, I think it's best to fix existing errors first.

### CODE REVIEW

File: src/polyfills/registerPolyfill.ts [hunk 1]
```diff
@@ -5,10 +5,10 @@ import { patternToRegex } from "webext-patterns";
 import { browser, WebNavigation } from "webextension-polyfill-ts";
 import OnDOMContentLoadedDetailsType = WebNavigation.OnDOMContentLoadedDetailsType;
 
-// @ts-expect-error
+// @ts-expect-error Will be dropped by #535
 async function p<T>(fn, ...args): Promise<T> {
   return new Promise((resolve, reject) => {
-    // @ts-expect-error
+    // @ts-expect-error Will be dropped by #535
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/polyfills/registerPolyfill.ts:11]: I merged in #535

File: src/options/pages/brickEditor/CodeEditor.tsx [hunk 2]
```diff
@@ -22,20 +22,13 @@ import React, { useState, useRef, Suspense, useEffect } from "react";
...
-  .default;
-const blueprintTemplate = require("raw-loader!@contrib/templates/blueprint-menu.txt?esModule=false")
-  .default;
+import serviceTemplate from "@contrib/templates/service.txt";
+import emberjsTemplate from "@contrib/templates/reader-emberjs.txt";
+import jqueryTemplate from "@contrib/templates/reader-jquery.txt";
+import reactTemplate from "@contrib/templates/reader-react.txt";
+import menuTemplate from "@contrib/templates/foundation-menu-item.txt";
+import panelTemplate from "@contrib/templates/foundation-panel.txt";
+import blueprintTemplate from "@contrib/templates/blueprint-menu.txt";
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/options/pages/brickEditor/CodeEditor.tsx:30]: @fregante You can test this by 1) going to the workshop, 2) clicking create new brick, and 3) selecting a template from the dropdown

See documentation for more information on accessing: https://docs.pixiebrix.com/developer-guide/workshop-basics#202afa4a0c0b4c449031d30540ba9a5d

**Comment by twschiller** [comment]
@fregante see comment on how to test. 

Also, I merged in the prettier PR you made, so you might need to re-run on these files? Do we need to fix the prettier configuration in pre-commit: http://github.com/pixiebrix/pixiebrix-extension/blob/7e46cde8f20cc265af7668d1d40c98a2bab9160a/.pre-commit-config.yaml#L13-L13?

File: .github/workflows/integration.yml [hunk 3]
```diff
@@ -63,3 +63,20 @@ jobs:
...
+        with:
+          node-version: '14.x'
+      - uses: bahmutov/npm-install@v1
+        # Run eslint without GitHub Actions annotations
+        # https://stackoverflow.com/a/65964721/288906
+      - name: npm run lint
+        run: |
+          echo "::remove-matcher owner=eslint-compact::"
+          echo "::remove-matcher owner=eslint-stylish::"
+          npm run lint
```
[the remark this answers is not in the document]
Comment (fregante) [reply .github/workflows/integration.yml:82]: The workflow would be this, if interested, but I think it's best to just leave it running at all times. It costs nothing and it already passes thanks to this PR 🎉 

<details>
<summary>lint.yml</summary>


```yml
name: Lint

on:
  push:
    branches:
      - "*"
    path:
      - .eslintrc
      - package-lock.json

jobs:
  lint:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v1
        with:
          node-version: "14.x"
      - uses: bahmutov/npm-install@v1

      # Run eslint without GitHub Actions annotations
      # https://stackoverflow.com/a/65964721/288906
      - name: npm run lint
        run: |
          echo "::remove-matcher owner=eslint-compact::"
          echo "::remove-matcher owner=eslint-stylish::"
          npm run lint
```

</details>

File: src/options/pages/brickEditor/CodeEditor.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (fregante) [reply src/options/pages/brickEditor/CodeEditor.tsx:30]: Thanks! With your steps I'll find my way around the code base over time. As you said going from the UI to the code is as easy as searching a string, but code -> UI takes a bit more time.

**Comment by fregante** [comment]
Prettier run, PR ready to merge.

**Review by twschiller — APPROVED** [review]

**Pull request merged**